subsys/bluetooth/mesh/Kconfig

```diff
@@ -427,10 +436,17 @@ config BT_MESH_RX_SEG_MAX
 	  which leaves 56 bytes for application layer data using a
 	  4-byte MIC and 52 bytes using an 8-byte MIC.
 
+config BT_MESH_RX_SEG_MAX
+	# Hidden option if BT_MESH_RX_SEG_MSG_COUNT set zero.
+	int
+	default 0
```

Comment:
It took some time to figure out what this is trying to do. I haven't seen such shadowing of options used elsewhere (at least within a single Kconfig file). Does it actually work without any warnings?

I'd prefer if we'd stick to the same convention used in other places of the stack for such situations, i.e. something like:
```c
#if defined(CONFIG_BT_MESH_RX_SEG_MAX)
#define RX_SEG_MAX CONFIG_BT_MESH_RX_SEG_MAX
#else
#define RX_SEG_MAX 0
#endif
```